languagetool-language-modules/ar/src/main/java/org/languagetool/tagging/ar/ArabicTagger.java

```diff
@@ -0,0 +1,207 @@
+        l.add(new AnalyzedToken(word, null, null));
+      }
+      tokenReadings.add(new AnalyzedTokenReadings(l, pos));
+      pos += word.length();
+    }
+    return tokenReadings;
+  }
+
+
+  @Nullable
+  protected List<AnalyzedToken> additionalTags(String word, IStemmer stemmer) {
+    List<AnalyzedToken> additionalTaggedTokens = new ArrayList<>();
+
+    //Any well-formed word started by conjuction letters like Waw and Feh is the same word without  the conjuction + a conjuction tag
+    // case of Lam jar + Al ta3rif
+    // w + ll or f + ll
+    List<String> tags = new ArrayList();
+    List<String> tags_with_jar = new ArrayList();
+    String possibleWord = word;
+    if (word.startsWith("ولل") || word.startsWith("فلل")) {
+      possibleWord = possibleWord.replaceAll("^[وف]لل", "بال");
+      tags_with_jar.add("L"); // only if has_jar
+      tags_with_jar.add("W"); // only if has_jar
+    } else if (word.startsWith("وك") || word.startsWith("فك") || word.startsWith("ول") || word.startsWith("فل")) {
+      possibleWord = possibleWord.replaceAll("^[وف][كل]", "ب");
+      if (word.startsWith("وك") || word.startsWith("فك")) {
+        tags_with_jar.add("K"); // only if has_jar
+      }
+      if (word.startsWith("ول") || word.startsWith("فل")) {
+        tags_with_jar.add("L"); // only if has_jar
+      }
+      tags_with_jar.add("W"); // only if has_jar
+    } else if (word.startsWith("و") || word.startsWith("ف")) {
+      possibleWord = possibleWord.replaceAll("^[وف]", "");
+      tags.add("W");
+
+    } else if (word.startsWith("لل") || word.startsWith("لل")) {
+      possibleWord = possibleWord.replaceAll("^لل", "بال");
+      tags_with_jar.add("L"); // only if has_jar
+    } else if (word.startsWith("ك")) {
+      possibleWord = possibleWord.replaceAll("^[ك]", "ب");
+      tags_with_jar.add("K"); // only if has_jar
+    } else if (word.startsWith("ل")) {
+      possibleWord = possibleWord.replaceAll("^[ل]", "ب");
+      tags_with_jar.add("L"); // only if has_jar
+    }
+
+    // to avoid redendecy all attached pronouns in dictionary are represented only by a generic pronoun
+    //  for words like بيتك بيتكما بيتهم بيتنا بيتكن there are one word which is بيتك
+    // we can simulate all word forms into the same tag which endec by H( H is tag for attached pronouns الضمائر المصتلة)
+    if (word.endsWith("ه")
+      || word.endsWith("ها")
+      || word.endsWith("هما")
+      || word.endsWith("كما")
+      || word.endsWith("هم")
+      || word.endsWith("هن")
+      || word.endsWith("كم")
+      || word.endsWith("كن")
+      || word.endsWith("نا")
+    ) {
+      possibleWord = possibleWord.replaceAll("(ه|ها|هما|هم|هن|كما|كم|كن|نا|ي)$", "ك");
+      List<AnalyzedToken> taggerTokens;
+      taggerTokens = asAnalyzedTokenList(possibleWord, stemmer.lookup(possibleWord));
+      for (AnalyzedToken taggerToken : taggerTokens) {
+        String posTag = taggerToken.getPOSTag();
+        if (hasFlag(posTag, "has_pronoun")) {
+        }
```

Comment:
There is nothing in the 'if' block.  Is something missing or should we remove it?